zetasql/parser/bison_parser.y

```diff
@@ -3375,27 +3382,45 @@ export_model_statement:
     ;
 
 grant_statement:
-    "GRANT" privileges "ON" identifier path_expression "TO" grantee_list
+    "GRANT" privileges "ON" grant_object_kind grant_path "TO" grantee_list opt_grant_option
       {
-        $$ = MAKE_NODE(ASTGrantStatement, @$, {$2, $4, $5, $7});
+        auto grant_statement = MAKE_NODE(ASTGrantStatement, @$, {$2, $5, $7});
+        grant_statement->set_object_kind($4);
+        grant_statement->set_is_with_grant_option($8);
+        $$ = grant_statement;
       }
-    | "GRANT" privileges "ON" path_expression "TO" grantee_list
+    | "GRANT" privileges "ON" grant_path "TO" grantee_list opt_grant_option
       {
-        $$ = MAKE_NODE(ASTGrantStatement, @$, {$2, $4, $6});
+        auto grant_statement = MAKE_NODE(ASTGrantStatement, @$, {$2, $4, $6});
+        grant_statement->set_is_with_grant_option($7);
+        $$ = grant_statement;
       }
     ;
 
 revoke_statement:
-    "REVOKE" privileges "ON" identifier path_expression "FROM" grantee_list
```

Comment:
can we just extend `"*"` rule here, just like how `select_column` did? the `grant_path` rule  looks strange

Reply:
what's the different with `select_column`

Reply:
try not change the corresponding class fields. `ASTIdentifier` can still handle "*" , handle "*" in `revoke_statement` or `grant_statement`

Reply:
please show your detailed design